Automattic/woocommerce-payments
# Add/e2e shopper subscriptions free trial

Closes https://github.com/Automattic/woocommerce-payments/issues/2221.

#### Changes proposed in this Pull Request

Title: E2E Test for the `Shopper / Subscriptions / Purchase free trial subscription` flow.

Description: This E2E test does the following steps:
1. Logs in as a shopper and signs up for a free trial subscription that has no sign up fees.
2. Logs in as the merchant and verifies:
    1. That "Setup intent" was created instead of "Payment intent".
    2. That the customer was not charged for the free trial.
    2. That the free trial details such as the free trial end date is correctly saved. 

#### Testing instructions

1. Run the newly added test:
    ```
    npm run test:e2e ./tests/e2e/specs/shopper/shopper-subscriptions-free-trial.spec.js
    ```
2. All these tests should pass:
    <img width="603" alt="Screen Shot 2021-07-31 at 2 22 41 AM" src="https://user-images.githubusercontent.com/4509348/127695838-cecad44d-fb6e-4606-8bae-1aad8aea5245.png">
